In [852]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

%matplotlib inline

In [312]:
df = pd.read_csv("C:\\Users\\Fabio\\Desktop\\Data Science\\banco_dados\\athlete_events.csv")

In [313]:
df.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [314]:
df.describe()

ID            Age         Height         Weight  \
count  271116.000000  261642.000000  210945.000000  208241.000000   
mean    68248.954396      25.556898     175.338970      70.702393   
std     39022.286345       6.393561      10.518462      14.348020   
min         1.000000      10.000000     127.000000      25.000000   
25%     34643.000000      21.000000     168.000000      60.000000   
50%     68205.000000      24.000000     175.000000      70.000000   
75%    102097.250000      28.000000     183.000000      79.000000   
max    135571.000000      97.000000     226.000000     214.000000   

                Year  
count  271116.000000  
mean     1978.378480  
std        29.877632  
min      1896.000000  
25%      1960.000000  
50%      1988.000000  
75%      2002.000000  
max      2016.000000

# História das Olimpíadas  
_(créditos ao prof. Rafael Moreira)_

Após um ano de atraso por conta da pandemia de Covid-19, as atenções do mundo todo se voltaram para Tóquio, no Japão, para acompanhar mais uma edição das Olimpíadas.

No Brasil não foi diferente, e muitos se uniram para torcer por nossos atletas em diferentes competições, tanto em esportes onde o Brasil já possui tradição quanto em novos esportes.

Vamos aproveitar o clima para estudar um pouco das Olimpíadas! Utilizaremos um _dataset_ com 120 anos de dados históricos das Olimpíadas, cobrindo desde os jogos de Atenas 1896 até Rio 2016. 

Faça o download do _dataset_ em https://www.kaggle.com/heesoo37/120-years-of-olympic-history-athletes-and-results e carregue o arquivo ```athlete_events.csv``` para um DataFrame utilizando Pandas. Aproveite para explorar seu DataFrame e se familiarizar com a sua estrutura. 

OBS: Fique à vontade para acrescentar mais células Python conforme necessário em qualquer etapa do exercício.

## 1. O Brasil nas Olimpíadas

Vamos começar estudando o desempenho do nossos próprio país. Gere um DataFrame novo contendo apenas as informações sobre atletas brasileiros.

In [315]:
df_atletas_brasileiros = df[df["Team"] == "Brazil"]
df_atletas_brasileiros.head()

ID                        Name Sex   Age  Height  Weight    Team  NOC  \
665   386             Alexandre Abeid   M  22.0   194.0    92.0  Brazil  BRA   
666   386             Alexandre Abeid   M  26.0   194.0    92.0  Brazil  BRA   
668   388  Abel Carlos da Silva Braga   M  19.0   190.0    73.0  Brazil  BRA   
781   451         Diana Monteiro Abla   F  21.0   175.0    75.0  Brazil  BRA   
1005  565        Glauclio Serro Abreu   M  26.0   185.0    75.0  Brazil  BRA   

            Games  Year  Season            City       Sport  \
665   1972 Summer  1972  Summer          Munich  Volleyball   
666   1976 Summer  1976  Summer        Montreal  Volleyball   
668   1972 Summer  1972  Summer          Munich    Football   
781   2016 Summer  2016  Summer  Rio de Janeiro  Water Polo   
1005  2004 Summer  2004  Summer          Athina      Boxing   

                              Event Medal  
665     Volleyball Men's Volleyball   NaN  
666     Volleyball Men's Volleyball   NaN  
668         Football Men's Football   NaN  
781   Water Polo Women's Water Polo   NaN  
1005      Boxing Men's Middleweight   NaN

### Medalhistas

Vamos focar um pouco nos casos de sucesso do Brasil. Use o seu DataFrame anterior para filtrar apenas informações sobre **medalhistas** brasileiros. 

**DICA:** observe como a coluna ```Medal``` é representada quando o atleta não ganhou medalha.

In [316]:
df_atletas_brasileiros_medalhistas = df_atletas_brasileiros.dropna(subset=["Medal"])

In [317]:
df_atletas_brasileiros_medalhistas.head()

ID                                 Name Sex   Age  Height  Weight  \
1651   918                  Ademir Roque Kaefer   M  24.0   179.0    74.0   
1652   918                  Ademir Roque Kaefer   M  28.0   179.0    74.0   
1668   925  Adenzia Aparecida Ferreira da Silva   F  25.0   187.0    65.0   
1733   966                         Daniel Adler   M  26.0   180.0    72.0   
1856  1020         Adriana Aparecida dos Santos   F  25.0   180.0    61.0   

        Team  NOC        Games  Year  Season         City       Sport  \
1651  Brazil  BRA  1984 Summer  1984  Summer  Los Angeles    Football   
1652  Brazil  BRA  1988 Summer  1988  Summer        Seoul    Football   
1668  Brazil  BRA  2012 Summer  2012  Summer       London  Volleyball   
1733  Brazil  BRA  1984 Summer  1984  Summer  Los Angeles     Sailing   
1856  Brazil  BRA  1996 Summer  1996  Summer      Atlanta  Basketball   

                                    Event   Medal  
1651              Football Men's Football  Silver  
1652              Football Men's Football  Silver  
1668        Volleyball Women's Volleyball    Gold  
1733  Sailing Mixed Three Person Keelboat  Silver  
1856        Basketball Women's Basketball  Silver

### Verão vs Inverno

Você deve ter notado que temos duas categorias distintas de jogos olímpicos, representados pela estação: temos os jogos de verão e os jogos de inverno, que ocorrem de maneira intercalada.

Agora que já conhecemos os medalhistas brasileiros, resposta: quantos atletas brasileiros receberam medalha nos jogos de verão e quantos receberam nos jogos de inverno?

In [318]:
jogos_verao_medalhistas = df_atletas_brasileiros_medalhistas["Season"].value_counts().sum()

In [319]:
print(f"{jogos_verao_medalhistas} atletas brasileiros receberam medalhas nos jogos de verão")

449 atletas brasileiros receberam medalhas nos jogos de verão


In [320]:
jogos_verao_medalhistas_por_medalhas = df_atletas_brasileiros_medalhistas.groupby(by=["Medal"])["Season"].value_counts()
print(f"Quadro de medalhas para atletas brasileiros nos jogos de verão: {pd.DataFrame(jogos_verao_medalhistas_por_medalhas)}")

Quadro de medalhas para atletas brasileiros nos jogos de verão:                Season
Medal  Season        
Bronze Summer     185
Gold   Summer     103
Silver Summer     161


Os jogos de verão são bem mais populares do que os jogos de inverno no Brasil. Portanto, deste ponto em diante iremos focar apenas nos jogos de verão. Descarte de seu DataFrame os dados dos jogos de inverno.



In [321]:
df_atletas_brasileiros_verao = df_atletas_brasileiros[df_atletas_brasileiros["Season"] == "Summer"]
df_atletas_brasileiros_verao.head()

ID                        Name Sex   Age  Height  Weight    Team  NOC  \
665   386             Alexandre Abeid   M  22.0   194.0    92.0  Brazil  BRA   
666   386             Alexandre Abeid   M  26.0   194.0    92.0  Brazil  BRA   
668   388  Abel Carlos da Silva Braga   M  19.0   190.0    73.0  Brazil  BRA   
781   451         Diana Monteiro Abla   F  21.0   175.0    75.0  Brazil  BRA   
1005  565        Glauclio Serro Abreu   M  26.0   185.0    75.0  Brazil  BRA   

            Games  Year  Season            City       Sport  \
665   1972 Summer  1972  Summer          Munich  Volleyball   
666   1976 Summer  1976  Summer        Montreal  Volleyball   
668   1972 Summer  1972  Summer          Munich    Football   
781   2016 Summer  2016  Summer  Rio de Janeiro  Water Polo   
1005  2004 Summer  2004  Summer          Athina      Boxing   

                              Event Medal  
665     Volleyball Men's Volleyball   NaN  
666     Volleyball Men's Volleyball   NaN  
668         Football Men's Football   NaN  
781   Water Polo Women's Water Polo   NaN  
1005      Boxing Men's Middleweight   NaN

### Atletas do Brasil

Vamos conhecer um pouco melhor nossos atletas. Descubra a altura e peso médio de nossos medalhistas.

In [322]:
altura_medio_brasileiro_medalhista_verao = df_atletas_brasileiros_medalhistas["Height"].mean()
print(f"A altura média dos brasileiros medalhistas é de {altura_medio_brasileiro_medalhista_verao:.2f} cm")

A altura média dos brasileiros medalhistas é de 182.36 cm


In [323]:
peso_medio_brasileiro_medalhista_verao = df_atletas_brasileiros_medalhistas["Weight"].mean()
print(f"A altura média dos brasileiros medalhistas é de {peso_medio_brasileiro_medalhista_verao:.2f} kg")

A altura média dos brasileiros medalhistas é de 76.62 kg


Imaginamos que diferentes esportes podem beneficiar diferentes tipos físicos, certo? Então refaça a análise anterior, mas obtendo os valores médios **por esporte**.

In [324]:
df_atletas_brasileiros_medalhistas.head(3)

ID                                 Name Sex   Age  Height  Weight  \
1651  918                  Ademir Roque Kaefer   M  24.0   179.0    74.0   
1652  918                  Ademir Roque Kaefer   M  28.0   179.0    74.0   
1668  925  Adenzia Aparecida Ferreira da Silva   F  25.0   187.0    65.0   

        Team  NOC        Games  Year  Season         City       Sport  \
1651  Brazil  BRA  1984 Summer  1984  Summer  Los Angeles    Football   
1652  Brazil  BRA  1988 Summer  1988  Summer        Seoul    Football   
1668  Brazil  BRA  2012 Summer  2012  Summer       London  Volleyball   

                              Event   Medal  
1651        Football Men's Football  Silver  
1652        Football Men's Football  Silver  
1668  Volleyball Women's Volleyball    Gold

Será que os dados acima influenciaram no interesse geral dos atletas pelo esporte ou realmente impactaram no desempenho deles? Podemos tentar descobrir se há algum tipo de correlação.

Você ainda possui o dataframe original contendo todos os atletas brasileiros, incluindo os sem medalha? Obtenha os valores médios de peso e altura por esporte daquele dataframe e compare-o com os dos medalhistas. Há alguma diferença significativa em algum esporte?

In [325]:
esporte_altura_medalhistas = df_atletas_brasileiros_medalhistas.groupby(by=["Sport"])["Height"].mean().sort_values()
print(f"Altura média por esporte com medalhas:(CM)\n{pd.DataFrame(esporte_altura_medalhistas)}")

Altura média por esporte com medalhas:(CM)
                       Height
Sport                        
Gymnastics         162.750000
Modern Pentathlon  166.000000
Boxing             170.000000
Canoeing           175.000000
Shooting           175.000000
Football           175.803030
Judo               176.666667
Equestrianism      179.666667
Athletics          181.000000
Sailing            181.588235
Taekwondo          184.000000
Basketball         185.607843
Swimming           189.105263
Volleyball         190.591667


In [326]:
df_atletas_brasileiros_altura = df_atletas_brasileiros.groupby(by=["Sport"])["Height"].mean().sort_values()
print(f"Altura média por esporte no total: (KG)\n{df_atletas_brasileiros_altura}")

Altura média por esporte no total: (KG)
Sport
Freestyle Skiing         154.000000
Figure Skating           156.000000
Gymnastics               157.268041
Rhythmic Gymnastics      164.160000
Snowboarding             167.000000
Diving                   167.444444
Trampolining             168.000000
Synchronized Swimming    168.461538
Modern Pentathlon        168.600000
Table Tennis             169.727273
Weightlifting            169.909091
Biathlon                 170.000000
Golf                     170.000000
Rugby Sevens             171.240000
Boxing                   171.985075
Archery                  172.600000
Cross Country Skiing     173.000000
Football                 173.369176
Taekwondo                173.769231
Judo                     173.925234
Shooting                 174.035294
Cycling                  174.235955
Luge                     175.000000
Badminton                175.500000
Triathlon                175.800000
Athletics                176.203463
Wrestling         

In [327]:
esporte_peso = df_atletas_brasileiros_medalhistas.groupby(by=["Sport"])["Weight"].mean().sort_values()
print(f"Peso médio por esporte de medalhistas: (KG)\n{pd.DataFrame(esporte_peso)}")

Peso médio por esporte de medalhistas: (KG)
                      Weight
Sport                       
Modern Pentathlon  55.000000
Gymnastics         63.750000
Boxing             64.000000
Shooting           69.000000
Football           69.961832
Athletics          74.576923
Equestrianism      75.000000
Basketball         78.478261
Taekwondo          79.500000
Sailing            80.411765
Volleyball         81.166667
Swimming           81.555556
Canoeing           83.250000
Judo               86.285714


In [328]:
df_atletas_brasileiros_peso = df_atletas_brasileiros.groupby(by=["Sport"])["Weight"].mean().sort_values()
print(f"Peso médio por esporte no total: (KG)\n{pd.DataFrame(df_atletas_brasileiros_peso)}")

Peso médio por esporte no total: (KG)
                          Weight
Sport                           
Figure Skating         47.000000
Rhythmic Gymnastics    48.160000
Freestyle Skiing       50.000000
Gymnastics             52.463918
Biathlon               55.000000
Synchronized Swimming  57.423077
Trampolining           58.000000
Modern Pentathlon      60.666667
Snowboarding           61.000000
Golf                   62.000000
Taekwondo              63.615385
Boxing                 64.106061
Cross Country Skiing   64.125000
Table Tennis           64.181818
Triathlon              64.400000
Diving                 64.970588
Cycling                66.870588
Football               67.760456
Athletics              67.795045
Archery                71.700000
Tennis                 72.218750
Equestrianism          72.313869
Rugby Sevens           72.880000
Fencing                73.357143
Badminton              74.000000
Swimming               74.586634
Alpine Skiing          74.789474
Luge 

Existe um detalhe importante passando batido até agora em nossa análise: as categorias esportivas costumam ser divididas por gênero justamente por conta de diferenças físicas entre homens e mulheres que poderiam influenciar no desempenho. Compare a altura e peso médios de atletas brasileiros por esporte segmentado por sexo.

In [329]:
df_atletas_brasileiros.head(3)

ID                        Name Sex   Age  Height  Weight    Team  NOC  \
665  386             Alexandre Abeid   M  22.0   194.0    92.0  Brazil  BRA   
666  386             Alexandre Abeid   M  26.0   194.0    92.0  Brazil  BRA   
668  388  Abel Carlos da Silva Braga   M  19.0   190.0    73.0  Brazil  BRA   

           Games  Year  Season      City       Sport  \
665  1972 Summer  1972  Summer    Munich  Volleyball   
666  1976 Summer  1976  Summer  Montreal  Volleyball   
668  1972 Summer  1972  Summer    Munich    Football   

                           Event Medal  
665  Volleyball Men's Volleyball   NaN  
666  Volleyball Men's Volleyball   NaN  
668      Football Men's Football   NaN

In [330]:
df_atletas_brasileiros_altura_media_esporte = df_atletas_brasileiros.groupby(by=["Sex", "Sport"])["Height"].mean()
df_atletas_brasileiros_altura_media_esporte

Sex  Sport        
F    Alpine Skiing    160.333333
     Archery          162.857143
     Athletics        167.492958
     Badminton        168.000000
     Basketball       182.865854
                         ...    
M    Triathlon        184.000000
     Volleyball       192.310976
     Water Polo       181.423729
     Weightlifting    171.777778
     Wrestling        182.142857
Name: Height, Length: 74, dtype: float64

In [331]:
df_atletas_brasileiros_peso_medio_esporte = df_atletas_brasileiros.groupby(by=["Sex", "Sport"])["Weight"].mean()
df_atletas_brasileiros_peso_medio_esporte

Sex  Sport        
F    Alpine Skiing    52.333333
     Archery          59.714286
     Athletics        59.809859
     Badminton        70.000000
     Basketball       73.381579
                        ...    
M    Triathlon        73.272727
     Volleyball       86.000000
     Water Polo       85.725000
     Weightlifting    90.000000
     Wrestling        97.285714
Name: Weight, Length: 74, dtype: float64

Qual foi (ou quais foram) o maior medalhista brasileiro em quantidade total de medalhas?

In [332]:
df_maiores_medalhista_brasileiro = df_atletas_brasileiros_medalhistas.groupby(by=["Name"])["Medal"].count().sort_values(ascending=False)
df_maiores_medalhista_brasileiro.head(10)


Name
Robert Scheidt                           5
Torben Schmidt Grael                     5
Srgio "Escadinha" Dutra dos Santos       4
Gustavo Frana Borges                     4
Rodrigo de Paula Pessoa                  3
Dante Guimares Santos do Amaral          3
Hlia "Fofo" Rogrio de Souza Pinto        3
Isaquias Queiroz dos Santos              3
Rodrigo "Rodrigo" Santana                3
Gilberto Amauri "Giba" de Godoy Filho    3
Name: Medal, dtype: int64

E o(s) maior(es) em quantidade de medalhas de ouro?

In [763]:
df_maiores_medalhista_brasileiro_ouro = df_atletas_brasileiros[df_atletas_brasileiros["Medal"] == "Gold"]
df_maiores_medalhista_brasileiro_ouro = df_maiores_medalhista_brasileiro_ouro.groupby(by=["Name"])["Medal"].count().sort_values(ascending=False)
df_maiores_medalhista_brasileiro_ouro.head(13)

Name
Marcelo Bastos Ferreira                               2
Thasa Daher de Menezes                                2
Adhemar Ferreira da Silva                             2
Maurcio Camargo Lima                                  2
Paula Renata Marques Pequeno                          2
Jaqueline Maria "Jaque" Pereira de Carvalho Endres    2
Giovane Farinazzo Gvio                                2
Robert Scheidt                                        2
Fabiana "Fabi" Alvim de Oliveira                      2
Sheilla Tavares de Castro Blassioli                   2
Srgio "Escadinha" Dutra dos Santos                    2
Fabiana Marcelino Claudino                            2
Torben Schmidt Grael                                  2
Name: Medal, dtype: int64

Qual esporte rendeu mais medalhas de ouro para o Brasil? E qual rendeu mais medalhas no total?

**DICA:** tome muito cuidado nessa análise: cada **evento esportivo** rende 1 medalha. Por exemplo, quando a equipe de futebol vence, isso é considerado 1 medalha, mesmo tendo cerca de 20 atletas medalhistas na equipe. 

In [397]:
df_atletas_brasileiros.head(3)

ID                        Name Sex   Age  Height  Weight    Team  NOC  \
665  386             Alexandre Abeid   M  22.0   194.0    92.0  Brazil  BRA   
666  386             Alexandre Abeid   M  26.0   194.0    92.0  Brazil  BRA   
668  388  Abel Carlos da Silva Braga   M  19.0   190.0    73.0  Brazil  BRA   

           Games  Year  Season      City       Sport  \
665  1972 Summer  1972  Summer    Munich  Volleyball   
666  1976 Summer  1976  Summer  Montreal  Volleyball   
668  1972 Summer  1972  Summer    Munich    Football   

                           Event Medal  
665  Volleyball Men's Volleyball   NaN  
666  Volleyball Men's Volleyball   NaN  
668      Football Men's Football   NaN

In [423]:
df_maiores_medalhista_brasileiro_ouro = df_atletas_brasileiros[df_atletas_brasileiros["Medal"] == "Gold"]
df_esporte_mais_medalha_ouro_brasil = df_maiores_medalhista_brasileiro_ouro.groupby(by=["Sport"])["Medal"].count().sort_values(ascending=False)
df_esporte_mais_medalha_ouro_brasil.head(10)

Sport
Volleyball       60
Football         17
Sailing          12
Athletics         5
Judo              4
Boxing            1
Equestrianism     1
Gymnastics        1
Shooting          1
Swimming          1
Name: Medal, dtype: int64

Cada "categoria" dentro de um esporte é considerado um evento. Por exemplo, dentro de "atletismo", temos uma competição de 100m masculina, uma de 100m feminino, um revezamento 4 x 100m masculino, um revezamento 4 x 100m feminino, uma competição de 400m masculino, uma de 400m feminino, uma maratona masculina, uma maratona feminina, e assim sucessivamente.

Sabendo disso, qual evento esportivo mais rendeu medalhas de ouro para o Brasil? E total de medalhas?

In [418]:
df_maiores_medalhista_brasileiro_ouro = df_atletas_brasileiros[df_atletas_brasileiros["Medal"] == "Gold"]
df_maiores_medalhista_brasileiro_ouro = df_maiores_medalhista_brasileiro_ouro.groupby(by=["Event"])["Medal"].count().sort_values(ascending=False)
df_maiores_medalhista_brasileiro_ouro.head(10)

Event
Volleyball Men's Volleyball          36
Volleyball Women's Volleyball        24
Football Men's Football              17
Sailing Men's Two Person Keelboat     2
Athletics Men's Triple Jump           2
Sailing Women's Skiff                 2
Sailing Mixed Two Person Keelboat     2
Sailing Mixed Two Person Dinghy       2
Sailing Mixed One Person Dinghy       2
Sailing Mixed Multihull               2
Name: Medal, dtype: int64

Para finalizar sobre o Brasil: obtenha o total de medalhas de ouro, prata, bronze e total por ano.

In [429]:
df_atletas_brasileiros_medalhistas.head(3)

ID                                 Name Sex   Age  Height  Weight  \
1651  918                  Ademir Roque Kaefer   M  24.0   179.0    74.0   
1652  918                  Ademir Roque Kaefer   M  28.0   179.0    74.0   
1668  925  Adenzia Aparecida Ferreira da Silva   F  25.0   187.0    65.0   

        Team  NOC        Games  Year  Season         City       Sport  \
1651  Brazil  BRA  1984 Summer  1984  Summer  Los Angeles    Football   
1652  Brazil  BRA  1988 Summer  1988  Summer        Seoul    Football   
1668  Brazil  BRA  2012 Summer  2012  Summer       London  Volleyball   

                              Event   Medal  
1651        Football Men's Football  Silver  
1652        Football Men's Football  Silver  
1668  Volleyball Women's Volleyball    Gold

In [432]:
df_atletas_brasileiros_medalhistas["Medal"].value_counts()

Bronze    185
Silver    161
Gold      103
Name: Medal, dtype: int64

In [428]:
df_atletas_brasileiros_total_medalhas = df_atletas_brasileiros_medalhistas.groupby(by=["Year"])["Medal"].value_counts()
df_atletas_brasileiros_total_medalhas

Year  Medal 
1920  Bronze     5
      Gold       1
      Silver     1
1948  Bronze    10
1952  Bronze     2
      Gold       1
1956  Gold       1
1960  Bronze    13
1964  Bronze    12
1968  Bronze     3
      Silver     1
1972  Bronze     2
1976  Bronze     3
1980  Bronze     5
      Gold       4
1984  Silver    33
      Bronze     2
      Gold       1
1988  Silver    17
      Bronze     5
      Gold       1
1992  Gold      13
      Silver     1
1996  Bronze    43
      Silver    13
      Gold       3
2000  Bronze    34
      Silver     8
2004  Silver    17
      Gold      16
      Bronze     3
2008  Silver    32
      Bronze    28
      Gold      14
2012  Silver    32
      Gold      14
      Bronze     9
2016  Gold      34
      Bronze     6
      Silver     6
Name: Medal, dtype: int64

## 2. O mundo nos jogos de verão

Vamos agora analisar um pouquinho do que aconteceu nas Olimpíadas de verão em todo o mundo.

Retome o DataFrame original e descarte as informações sobre os jogos de inverno.

Obtenha a lista de todos os esportes já disputados nas olimpíadas de verão.

In [450]:
np.array(jogos_verao["Sport"])

array(['Basketball', 'Judo', 'Football', ..., 'Rowing', 'Rowing',
       'Basketball'], dtype=object)

Obtenha a lista de todas as modalidades esportivas já disputadas nas olimpíadas de verão.

In [451]:
np.array(jogos_verao["Event"])

array(["Basketball Men's Basketball", "Judo Men's Extra-Lightweight",
       "Football Men's Football", ...,
       "Rowing Men's Lightweight Coxless Fours",
       "Rowing Men's Lightweight Coxless Fours",
       "Basketball Women's Basketball"], dtype=object)

In [440]:
jogos_verao = df[df["Season"] == "Summer"]
jogos_verao.head(3)

ID                 Name Sex   Age  Height  Weight     Team  NOC  \
0   1            A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   2             A Lamusi   M  23.0   170.0    60.0    China  CHN   
2   3  Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   

         Games  Year  Season       City       Sport  \
0  1992 Summer  1992  Summer  Barcelona  Basketball   
1  2012 Summer  2012  Summer     London        Judo   
2  1920 Summer  1920  Summer  Antwerpen    Football   

                          Event Medal  
0   Basketball Men's Basketball   NaN  
1  Judo Men's Extra-Lightweight   NaN  
2       Football Men's Football   NaN

Obtenha a lista de todos os países que já disputaram olimpíadas.

In [465]:
jogos_verao["Team"].unique()

array(['China', 'Denmark', 'Denmark/Sweden', ..., 'Dow Jones', 'China-3',
       'Digby'], dtype=object)

Qual atleta foi o maior medalhista (em medalhas totais) da história das olimpíadas de verão?

In [500]:
jogos_verao_maior_medalhista = jogos_verao.groupby(by=["Name"])["Medal"].count().sort_values(ascending=False)
jogos_verao_maior_medalhista.head(3)

Name
Michael Fred Phelps, II               28
Larysa Semenivna Latynina (Diriy-)    18
Nikolay Yefimovich Andrianov          15
Name: Medal, dtype: int64

Qual atleta foi o maior medalhista de ouro da história das olimpíadas de verão?

In [504]:
jogos_verao_maior_medalhista_ouro = jogos_verao.groupby(by=["Name"])["Medal"].value_counts().sort_values(ascending=False)
jogos_verao_maior_medalhista_ouro.head(3)

Name                         Medal
Michael Fred Phelps, II      Gold     23
Raymond Clarence "Ray" Ewry  Gold     10
Mark Andrew Spitz            Gold      9
Name: Medal, dtype: int64

Qual país foi o maior medalhista de ouro da história das olimpíadas de verão? Lembre-se da questão do evento esportivo, para não considerar múltiplas medalhas para um mesmo evento (ex: uma equipe de futebol fazendo parecer que mais de 20 medalhas foram distribuídas).

In [529]:
df_paises_medalha_ouro = df[df["Medal"] == "Gold"]
df_paises_medalha_ouro.head(3)

ID                     Name Sex   Age  Height  Weight            Team  \
3    4     Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
42  17  Paavo Johannes Aaltonen   M  28.0   175.0    64.0         Finland   
44  17  Paavo Johannes Aaltonen   M  28.0   175.0    64.0         Finland   

    NOC        Games  Year  Season    City       Sport  \
3   DEN  1900 Summer  1900  Summer   Paris  Tug-Of-War   
42  FIN  1948 Summer  1948  Summer  London  Gymnastics   
44  FIN  1948 Summer  1948  Summer  London  Gymnastics   

                               Event Medal  
3        Tug-Of-War Men's Tug-Of-War  Gold  
42  Gymnastics Men's Team All-Around  Gold  
44      Gymnastics Men's Horse Vault  Gold

In [528]:
df_pais_mais_medalha_ouro = df_paises_medalha_ouro.groupby(by=["Team"])["Medal"].count().sort_values(ascending=False)
df_pais_mais_medalha_ouro.head(3)

Team
United States    2474
Soviet Union     1058
Germany           679
Name: Medal, dtype: int64

Obtenha o total de medalhas de ouro, prata e total por edição das Olimpíadas de verão. Lembre-se da questão do evento esportivo.

In [570]:
df_ouro = df[df["Medal"] == "Gold"]
df_ouro = df_ouro.groupby(by=["Games"])["Medal"].value_counts()
df_ouro = pd.DataFrame(df_ouro)
df_ouro.head()

Medal
Games       Medal       
1896 Summer Gold      62
1900 Summer Gold     201
1904 Summer Gold     173
1906 Summer Gold     157
1908 Summer Gold     294

In [572]:
df_prata = df[df["Medal"] == "Silver"]
df_prata = df_prata.groupby(by=["Games"])["Medal"].value_counts()
df_prata = pd.DataFrame(df_prata)
df_prata.head()

Medal
Games       Medal        
1896 Summer Silver     43
1900 Summer Silver    228
1904 Summer Silver    163
1906 Summer Silver    156
1908 Summer Silver    281

In [583]:
df = df[df["Season"] == "Summer"]
df_ouro_prata_total = df.groupby(by=["Games"])["Medal"].count()
df_ouro_prata_total = pd.DataFrame(df_ouro_prata_total)
df_ouro_prata_total.head()

Medal
Games             
1896 Summer    143
1900 Summer    604
1904 Summer    486
1906 Summer    458
1908 Summer    831

Qual país foi o maior medalhista em medalhas totais na história das olimpíadas de verão?

In [434]:
df.head(3)

ID                 Name Sex   Age  Height  Weight     Team  NOC  \
0   1            A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   2             A Lamusi   M  23.0   170.0    60.0    China  CHN   
2   3  Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   

         Games  Year  Season       City       Sport  \
0  1992 Summer  1992  Summer  Barcelona  Basketball   
1  2012 Summer  2012  Summer     London        Judo   
2  1920 Summer  1920  Summer  Antwerpen    Football   

                          Event Medal  
0   Basketball Men's Basketball   NaN  
1  Judo Men's Extra-Lightweight   NaN  
2       Football Men's Football   NaN

In [584]:
df_verao = df[df["Season"] == "Summer"]

In [607]:
df_verao_pais_maior_medalhista = df_verao.groupby(by=["Team"])["Medal"].value_counts().sort_values(ascending=False)
df_verao_pais_maior_medalhista

Team           Medal 
United States  Gold      2333
               Silver    1241
               Bronze    1112
Soviet Union   Gold       830
               Silver     635
                         ... 
Cyprus         Silver       1
Cote d'Ivoire  Silver       1
               Gold         1
               Bronze       1
England        Silver       1
Name: Medal, Length: 734, dtype: int64

In [611]:
df_verao_pais_maior_medalhista[0:3].sum()

4686

## 3. Brasil vs Mundo

Para finalizar, vamos fazer algumas comparações entre Brasil e mundo. Qual o ranking do Brasil em cada edição das olimpíadas? Lembrando que o ranking é ordenado por medalhas de ouro.

In [666]:
df_verao.head(3)

ID                 Name Sex   Age  Height  Weight     Team  NOC  \
0   1            A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   2             A Lamusi   M  23.0   170.0    60.0    China  CHN   
2   3  Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   

         Games  Year  Season       City       Sport  \
0  1992 Summer  1992  Summer  Barcelona  Basketball   
1  2012 Summer  2012  Summer     London        Judo   
2  1920 Summer  1920  Summer  Antwerpen    Football   

                          Event Medal  
0   Basketball Men's Basketball   NaN  
1  Judo Men's Extra-Lightweight   NaN  
2       Football Men's Football   NaN

In [761]:
df_verao_ranking_brasil = df_verao.groupby(by=["Medal", "Games"])["Team"].value_counts().sort_index(ascending=True)
df_verao_ranking_brasil = pd.DataFrame(df_verao_ranking_brasil)
df_verao_ranking_brasil.head(20)

Team
Medal  Games       Team                               
Bronze 1896 Summer Australia/Great Britain           2
                   Austria                           2
                   Denmark                           3
                   Ethnikos Gymnastikos Syllogos     2
                   France                            2
                   Germany                           2
                   Great Britain                     2
                   Greece                           18
                   Hungary                           3
                   United States                     2
       1900 Summer A North American Team             4
                   Australia                         3
                   Austria                           3
                   Baby-1                            1
                   Bagatelle Polo Club, Paris        4
                   Belgium                           4
                   Bohemia                           1
                   Bohemia/Great Britain             2
                   Canada                            1
                   Crabe II-4                        5

In [ ]:
df_verao_ranking_brasil


Compare o maior medalhista em ouros do Brasil com o maior medalhista em ouros do mundo.

In [770]:
df_maiores_medalhista_brasileiro_ouro.head(5)

Name
Marcelo Bastos Ferreira         2
Thasa Daher de Menezes          2
Adhemar Ferreira da Silva       2
Maurcio Camargo Lima            2
Paula Renata Marques Pequeno    2
Name: Medal, dtype: int64

In [773]:
jogos_verao_maior_medalhista_ouro.head(1)

Name                     Medal
Michael Fred Phelps, II  Gold     23
Name: Medal, dtype: int64

Compare o maior medalhista em total de medalhas do Brasil com o maior medalhista em total de medalhas do mundo.

In [777]:
df_maiores_medalhista_brasileiro.head(2)

Name
Robert Scheidt          5
Torben Schmidt Grael    5
Name: Medal, dtype: int64

In [784]:
jogos_verao_maior_medalhista.head(1)

Name
Michael Fred Phelps, II    28
Name: Medal, dtype: int64

Compare o maior medalhista em ouros do Brasil com o maior medalhista do mundo no mesmo esporte.

In [861]:
df_maiores_medalhista_brasileiro_ouro.head(1)

Name
Marcelo Bastos Ferreira    2
Name: Medal, dtype: int64

In [805]:
df[df["Name"] == "Marcelo Bastos Ferreira"]["Sport"]

68550    Sailing
68551    Sailing
68552    Sailing
68553    Sailing
Name: Sport, dtype: object

In [860]:
maior_medalhista_sailing = df_verao[df_verao["Sport"] == "Sailing"].value_counts()
maior_medalhista_sailing = pd.DataFrame(maior_medalhista_sailing).reset_index()
maior_medalhista_sailing = maior_medalhista_sailing.groupby(by=["Name", "Sport"])["Medal"].value_counts().sort_values(ascending=False)
maior_medalhista_sailing.head(2)

Name                            Sport    Medal
Charles Benedict "Ben" Ainslie  Sailing  Gold     4
Paul Bert Elvstrm               Sailing  Gold     4
Name: Medal, dtype: int64

Compare o maior medalhista em total de medalhas do Brasil com o maior medalhista do mundo no mesmo esporte.

In [869]:
df_maiores_medalhista_brasileiro.head(2)

Name
Robert Scheidt          5
Torben Schmidt Grael    5
Name: Medal, dtype: int64

In [870]:
maior_medalhista_sailing.head(2)

Name                            Sport    Medal
Charles Benedict "Ben" Ainslie  Sailing  Gold     4
Paul Bert Elvstrm               Sailing  Gold     4
Name: Medal, dtype: int64

Calcule o percentual de medalhas de ouro, prata e bronze que o Brasil ganhou em cada olimpíada.

In [884]:
percentual_medalhas_olimpiada = df_atletas_brasileiros.groupby(by=["Games"])["Medal"].value_counts(normalize=True)*100
percentual_medalhas_olimpiada

Games        Medal 
1920 Summer  Bronze     71.428571
             Gold       14.285714
             Silver     14.285714
1948 Summer  Bronze    100.000000
1952 Summer  Bronze     66.666667
             Gold       33.333333
1956 Summer  Gold      100.000000
1960 Summer  Bronze    100.000000
1964 Summer  Bronze    100.000000
1968 Summer  Bronze     75.000000
             Silver     25.000000
1972 Summer  Bronze    100.000000
1976 Summer  Bronze    100.000000
1980 Summer  Bronze     55.555556
             Gold       44.444444
1984 Summer  Silver     91.666667
             Bronze      5.555556
             Gold        2.777778
1988 Summer  Silver     73.913043
             Bronze     21.739130
             Gold        4.347826
1992 Summer  Gold       92.857143
             Silver      7.142857
1996 Summer  Bronze     72.881356
             Silver     22.033898
             Gold        5.084746
2000 Summer  Bronze     80.952381
             Silver     19.047619
2004 Summer  Silver     47.2